sklearn/cluster/_birch.py

```diff
@@ -333,15 +333,16 @@ def radius(self):
 
 
 class Birch(ClusterMixin, TransformerMixin, BaseEstimator):
-    """Implements the BIRCH clustering algorithm.
+    """Implements a variant of the BIRCH clustering algorithm.
 
-    It is a memory-efficient, online-learning algorithm provided as an
+    BIRCH is a memory-efficient, online-learning algorithm provided as an
     alternative to :class:`MiniBatchKMeans`. It constructs a tree
     data structure with the cluster centroids being read off the leaf.
     These can be either the final cluster centroids or can be provided as input
     to another clustering algorithm such as :class:`AgglomerativeClustering`.
 
-    Read more in the :ref:`User Guide <birch>`.
```

Comment:
We still should have something saying "read more" as the user guide ordinarily describes the algorithm and its application. "Read more, including how this implementation differs from the original publication", perhaps?

Reply:
Sounds good to me.